include/seastar/net/tls.hh

```diff
@@ -263,6 +263,7 @@ namespace tls {
     class reloadable_credentials_base;
 
     using reload_callback = std::function<void(const std::unordered_set<sstring>&, std::exception_ptr)>;
+    using reload_callback_with_creds = std::function<void(const std::unordered_set<sstring> &, const certificate_credentials&, std::exception_ptr)>;
```

Comment:
Will we have reload_callback_with_creds_and_something_else? If there's any likelihood of that, let's replace the parameter with a struct that can be expanded. The struct can also serve to document the parameters.

Reply:
Is it worth that to break the API (will require scylla modification). Or new, struct + old?
Strictly, the callback arg should be a variant of { ep, <cert_data>+ whatnot }

Reply:
>will we have reload_callback_with_creds_and_something_else?

From the redpanda side, I doubt it. If it seems likely/desirable in a general sense, sure I can add it into this PR.

Reply:
I'm asking you, not about redpanda requirements, but whether there's more such auxiliary information that's likely to be added in the future. I'm not an expert about tls.

Reply:
Given that the callback is on a certificate_credentials, giving the object itself back is probably as much info as can be provided. I.e. I cannot see anything else that could be provided...